client/events.go

```diff
@@ -46,3 +47,48 @@ func (cli *Client) Events(ctx context.Context, options types.EventsOptions) (io.
 	}
 	return serverResponse.body, nil
 }
+
+// EventsWithHandler decodes the stream of events handing them off to the appropriate
+// handlers for processing. It's up to the caller to stop the processing by canceling
+// the context. The error channel will be closed once the end of the input stream has been
+// reached. If an error is received all processing will be stopped.
+func (cli *Client) EventsWithHandler(ctx context.Context, options types.EventsOptions, handler eventtypes.EventHandler) <-chan error {
```

Comment:
> Can I replace the legacy method with this new one? I figured we needed backwards compatibility so I went with another method name.

All the users of the `engine-api` vendor the dependency, so please feel free to change.

As it is, the method is very hard to use, so replacing the definition is fine. We'll stabilize these eventually, but we need to have an exported API worth stabilizing and we're not there yet.